### PR TITLE
Add wca_id phrase

### LINKED ISSUE #301: Add "wcaid" phrase
Hi again! Today I'm here to post another small suggestion. This time, it's about the "**WCA ID**" phrase in the FMC sheet.
In some languages, the noun phrases are inverted comparing to English. In this case, in my Vietnamese language for example, "**WCA ID**" is "**ID WCA**", and the FMC sheet should look like:
sample
(I don't know if you noticed or not, but I changed the font to **Segoe UI**.)

**TL; DR:** I suggest you to make a new small phrase called **wcaid**.
Thanks.

### LINKED ISSUE #307: Make the form's lines gray
Hi! I saw in the FMC sheet, the form's lines is 100% black, and that can make the text written there harder to read. So I want to make these lines gray.
I am going to bundle this with **add-wca-id** pull request.
Thanks.

### PR DESCRIPTION
**Pull request by leduyquang753**
This pull request adds the "wca_id" phrase and closes #301.
**I couldn't squash the commits, because after rebasing I was forced to merge, and the whole thing started being messy again. GAVE. UP.**
Thank you.

### CODE REVIEW

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 1]
```diff
@@ -711,11 +718,19 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
...
         cb.beginText();
+        cb.setColorFill(new BaseColor(0x00, 0x00, 0x00));
         cb.setFontAndSize(bf, fitText(new Font(bf), text, new Rectangle((int)rect.getRight(), (int)rect.getTop()), maxFontSize, false, 1));
         cb.showTextAligned(align, text, (int)rect.getLeft(), (int)rect.getBottom(), 0);
         cb.endText();
+        if (textBlurred != "") {
+            cb.beginText();
+            cb.setColorFill(new BaseColor(0xC6, 0xC6, 0xC6));
+            cb.setFontAndSize(bf, fitText(new Font(bf), text, new Rectangle((int)rect.getRight(), (int)rect.getTop()), maxFontSize, false, 1));
+            cb.showTextAligned(align, textBlurred, (int)rect.getLeft() + bf.getWidthPoint(text, fitText(new Font(bf), text, new Rectangle((int)rect.getRight(), (int)rect.getTop()), maxFontSize, false, 1)), (int)rect.getBottom(), 0);
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Are you calling `fitText` twice here? Could we DRY this up?

**Comment by leduyquang753** [comment]
**I GAVE UP!** After rebasing to squash the commits, GitHub forced me to pull and merge. And as the result, the whole thing became a mess. Please forgive me... Thanks...

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 2]
```diff
@@ -664,6 +665,96 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
...
+            }
+        }
+
+        // Center
+        float tableOffset = (tableWidth-(columns-1)*cellWidth-maxFirstColumnWidth)/2;
+
+        // Position the table
+        table.writeSelectedRows(0, -1, left-tableOffset, scrambleBorderTop+tableHeight, cb);
+
+        // Rules
```
Comment (leduyquang753) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Is this the update? I have just copy-pasted from the original code to revert the gray stuff and...

**Comment by jfly** [comment]
@leduyquang753, if you're done with this PR, can you close it?

**Comment by leduyquang753** [comment]
Uh, no no no no no, I meant I gave up with squashing the commits... Sorry for not being clear... The thing is still keeping going on.

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 3]
```diff
@@ -733,6 +732,9 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
             table.addCell(cell);
 
             for (int j=0; j<directionModifiers.length; j++) {
+                maxFirstColumnWidth = Math.max(maxFirstColumnWidth, bf.getWidthPoint(direction[j], firstColumnFontSize));
+
+                cell = new PdfPCell(new Phrase(direction[j], new Font(bf, firstColumnFontSize)));
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Is this supposed to be here? It looks like a duplicate of the next two lines.

Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Is this supposed to be here? It looks like a duplicate of the next two lines.

**Comment by leduyquang753** [comment]
Wait what? I didn't add those lines... @_@
Let me fix this...

**Comment by leduyquang753** [comment]
Oh, I see. Previously there was some changes to your original code, as tye result my old code had some conflicts, so I tried to resolve it. Accidentally I forgot to delete one copy of the conflicted code and yea, you got the idea... I can't edit the file now because I have school, but maybe I will do this at noon. Thanks for prompting me that.

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 4]
```diff
@@ -727,21 +727,13 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
 
             maxFirstColumnWidth = Math.max(maxFirstColumnWidth, bf.getWidthPoint(movesType[i], firstColumnFontSize));
 
-            PdfPCell cell = new PdfPCell(new Phrase(movesType[i], new Font(bf, firstColumnFontSize, Font.BOLD)));
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Is this supposed to change? Or another merge issue? =(

**Comment by leduyquang753** [comment]
OMG yes. There were two conflicts... And yesterday there was one more... Oh my... :[

**Comment by leduyquang753** [comment]
In order to completely fix these annoying stuff, I will wait for the other stuff related to this **scrambleRequest** file to finish themselves before re-copying its code and doing the changes. **Or if you want this to be finished first, PLEASE TELL ME!**. Thanks.

**Comment by jfly** [comment]
@leduyquang753, it's fine to wait on this.

**Comment by jfly** [comment]
@leduyquang753, I'm going to close this PR. Please feel free to reopen or create a new issue if you want to revisit this!

**Pull request closed without merging**